core/modules/layout/layout.theme.inc

```diff
@@ -345,34 +345,44 @@ function layout_preprocess_layout(&$variables) {
     $variables['row_data'] = array();
     foreach ($variables['rows'] as $name => $row) {
       $container = ($row['container'] == 'container') ? 'container container-fluid' : (($row['container'] == 'container_fluid') ? 'container-fluid' : 'no-container');
-      $row['row_class'] = $container . ' flexible-row--' . $name . ' ' . $row['contains'] . ' ' . $row['classes'];
+      $row['row_class'] = $container . ' flexible-row--' . $name . ' ' . $row['contains'] . ' ' . _layout_clean_custom_css($row['classes']);
       $row['row_id'] = $variables['flexible_editor'] ? 'id = "flexible-row--' . $name . '"' : '';
       $row['element'] = !empty($row['element']) ? $row['element'] : 'div';
       if ($row['contains'] == 'region_12') {
-        if (!empty($row['region_names']['region_name_0'])) {
-          $region_name = $row['region_names']['region_name_0'];
+        $region_classes = '';
+        if (!empty($row['region_names']['region_0'])) {
+          $region_name = $row['region_names']['region_0']['region_name'];
+          $region_classes .= 'l-region--1';
+          $region_classes .= isset($row['region_names']['region_0']['classes']) && !empty($row['region_names']['region_0']['region_class_enable']) ? ' ' . _layout_clean_custom_css($row['region_names']['region_0']['classes']) : '';
         }
         else {
           $region_name = $name;
+          $region_classes .= 'l-region--1';
         }
         $row['regions'][0]['region_md'] = '12';
         $row['regions'][0]['region_name'] = $region_name;
+        $row['regions'][0]['region_classes'] = $region_classes;
         $row['regions'][0]['content_key'] = $variables['flexible_editor'] ? $name : $name . '--0';
       }
       else {
         $col_info = $variables['column_data'][$row['contains']];
         $split = explode(':', $col_info['bootstrap']);
         $i = 0;
         foreach ($split as $col) {
-          if (!empty($row['region_names']['region_name_' . $i])) {
-            $region_name = $row['region_names']['region_name_' . $i];
+          $region_classes = '';
+          if (!empty($row['region_names']['region_' . $i])) {
+            $region_name = $row['region_names']['region_' . $i]['region_name'];
+            $region_classes .= 'l-region--' . ($i + 1);
+            $region_classes .= isset($row['region_names']['region_' . $i]['classes']) && !empty($row['region_names']['region_' . $i]['region_class_enable']) ? ' ' . _layout_clean_custom_css($row['region_names']['region_' . $i]['classes']) : '';
```

Comment:
@herbdool you seem to have a soft spot for really, really long lines. :wink: Slightly shorter would be a lot easier to read (and understand).